sdk/spring/scripts/ci_update_versions.py

```diff
@@ -0,0 +1,104 @@
+# Python version 3.4 or higher is required to run this script.
+#
+# Change the external_dependencies in sdk/spring for test compatibility of spring-boot version .
+#
+# How to use:
+# 1. Make sure file(`.\sdk\spring\spring_boot_SPRING_BOOT_VERSION_managed_external_dependencies.txt`) exist. If it doesn't exist, please run
+#    `.\sdk\spring\scripts\get_spring_boot_managed_external_dependencies.py` to create that file.
+# 2. Change `SPRING_BOOT_VERSION` in ci.yml for this script manually.
+# 3. Then the ci will automatically run command `python .\sdk\spring\scripts\ci_update_versions.py --sbv SPRING_BOOT_VERSION`.
+#
+# The script must be run at the root of azure-sdk-for-java.
+
+import in_place
+import argparse
+import os
+import re
+import time
+from log import log
+
+include_update_marker = re.compile(r'\{x-include-update;([^;]+);([^}]+)\}')
+version_update_marker = re.compile(r'\{x-version-update;([^;]+);([^}]+)\}')
+external_dependency_include_regex = r'(?<=<include>).+?(?=</include>)'
+external_dependency_version_regex = r'(?<=<version>).+?(?=</version>)'
+
+
+def update_versions(external_dependency_version_map, target_file):
+    # replace artifact versions in target_file
+    with in_place.InPlace(target_file) as file:
+        for line in file:
+            if line.startswith('#') or not line:
+                file.write(line)
+            elif version_update_marker.search(line):
+                match = version_update_marker.search(line)
+                module_name, version_type = match.group(1), match.group(2)
+                if module_name in external_dependency_version_map and version_type == 'external_dependency':
+                    new_version = external_dependency_version_map[module_name]
+                    newline = re.sub(external_dependency_version_regex, new_version, line)
+                    file.write(newline)
+                    log.info("updating " + target_file + ":" + newline)
+                else:
+                    file.write(line)
+            elif include_update_marker.search(line):
```

Comment:
It's possible that `version_update_marker.search(line) == True && include_update_marker.search(line) == True.`, so it should not be `elif`.

Reply:
They are two different conditions that need different processing methods,   ``version_update_marker.search(line)`` will be replace with ``external_dependency_version_map[module_name]`` but ``include_update_marker.search(line)`` will be replace with ``module_name + ':[' + external_dependency_version_map[module_name] + ']'`` , so I think they shouldn't be written together or connected with '&&'.